packages/sourcecred/src/core/ledger/ledger.js

```diff
@@ -847,6 +921,80 @@ export class Ledger {
     return this._grainIntegrationStatuses.get(id);
   }
 
+  /**
+   *   This function can be called while accounting is disabled to update
+   *   the ledger's external currency configuration. Note that this will also
+   *   deactivate identities missing a payout address that matches the new
+   *   configuration when accounting is disabled.
+   */
+  setExternalCurrency(chainId: ChainId, tokenAddress?: EthAddress): Ledger {
+    const currency = buildCurrency(chainId, tokenAddress);
+
+    if (!deepEqual(currency, this._externalCurrency)) {
+      this._createAndProcessEvent({
+        type: "SET_EXTERNAL_CURRENCY",
+        currency,
+      });
+    }
+
+    // no-op if already disabled
+    return this;
+  }
+
+  _setExternalCurrency({currency}: SetExternalCurrency) {
+    this._externalCurrency = currency;
+
+    // Warning: Account Mutations Below
+    if (!this._balanceAccountingEnabled) {
+      // Deactivate accounts if they don't have an address stored for
+      // the newly configured currency.
+      this._deactivateAccounts();
+    }
+  }
+
+  /**
+   * Cannot be invoked if accounting is still disabled
+   */
+  removeExternalCurrency(): Ledger {
+    if (this._externalCurrency) {
+      this._createAndProcessEvent({
+        type: "REMOVE_EXTERNAL_CURRENCY",
+      });
+    }
+    return this;
+  }
+
+  _removeExternalCurrency(_: RemoveExternalCurrency) {
+    if (!this._balanceAccountingEnabled) {
+      throw new Error(
+        "Cannot remove the external currency while Accounting is Disabled"
+      );
+    }
+    delete this._externalCurrency;
+  }
+
+  /**
+   * Returns the state of the ledger's accounting configuration. If accounting
+   * is enabled, currency will be undefined. if accounting is disabled, a
+   * Currency will be set.
+   */
+  accounting(): AccountingStatus {
+    return {
+      enabled: this._balanceAccountingEnabled,
+      trackDistributions: this._shouldTrackGrainIntegration,
+      currency: this._externalCurrency,
+    };
+  }
+
+  /**
+   * Expectably returns the currency from the
+   * externalCurrency prop. It can only be invoked when a currency is set.
```

Comment:
It can be invoked either way.
```suggestion
   * externalCurrency prop. Throws if a currency is not set.
```